src/main/asciidoc/java/http.adoc

```diff
@@ -590,6 +656,93 @@ vertx.createHttpServer().requestHandler(request -> {
 }).listen(8080);
 ----
 
+==== Writing HTTP/2 frames
+
+HTTP/2 is a framed protocol with various frames for the HTTP request/response model. The protocol allows other kind
+of frames to be sent and received.
+
+To send such frames, you can use the `link:../../apidocs/io/vertx/core/http/HttpServerResponse.html#writeFrame-int-int-io.vertx.core.buffer.Buffer-[writeFrame]` on the response.
+Here’s an example:
+
+[source,java]
+----
+int frameType = 40;
+int frameStatus = 10;
+Buffer payload = Buffer.buffer("some data");
+
+// Sending a frame to the client
+response.writeFrame(frameType, frameStatus, payload);
+----
+
+These frames are sent immediatly and are not subject to flow control - when such frame is sent there it may be done
+before other `DATA` frames.
+
+==== Stream reset
+
+HTTP/1.x does not allow a clean reset of a request or a response stream, for example when a client uploads
+a resource already present on the server, the server needs to accept the entire response.
+
+HTTP/2 supports stream reset at any time during the request/response:
+
+[source,java]
+----
+request.response().reset();
+----
+
+By default the `NO_ERROR` (0) error code is sent, another code can sent instead:
+
+[source,java]
+----
+request.response().reset(8);
+----
+
+The HTTP/2 specification defines the list of http://httpwg.org/specs/rfc7540.html#ErrorCodes[error codes] one can use.
+
+The request handler are notified of stream reset events with the `link:../../apidocs/io/vertx/core/http/HttpServerRequest.html#exceptionHandler-io.vertx.core.Handler-[request handler]` and
+`link:../../apidocs/io/vertx/core/http/HttpServerResponse.html#exceptionHandler-io.vertx.core.Handler-[response handler]`:
+
+[source,java]
+----
+request.response().exceptionHandler(err -> {
+  if (err instanceof StreamResetException) {
+    StreamResetException reset = (StreamResetException) err;
+    System.out.println("Stream reset " + reset.getCode());
+  }
+});
+----
+
+==== Server push
+
+Server push is a new feature of HTTP/2 that enables sending multiple responses in parallel for a single client request.
+
+When a server process a request, it can push a request/response to the client:
+
+[source,java]
+----
+HttpServerResponse response = request.response();
```

Comment:
it is async because the client may have limited the number of streams a server can send in the settings when establishing the connection, when it happens the push are queued.

the server must push streams before it ends the current response.

it the example:
- handler calls push : this sends a _push promise_ frame to the client
- handler calls send file : it sends the _headers_ frame + the _data_ frames to the client
- the push handler is called, it will send the headers + data frames of `main.js` to the client

probably that data frames of the original response and `main.js` are interleaved and multiplexed.